src/ngraph/pass/nop_elimination.cpp

```diff
@@ -134,62 +134,235 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node)
     return false;
 }
 
+static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node)
+{
+    auto out_shape = node->get_output_partial_shape(0);
+    if (!out_shape.is_dynamic() && shape_size(out_shape.get_shape()) != 0)
+    {
+        shared_ptr<Node> reshape;
+        auto input = node->input_value(0).get_node_shared_ptr();
+        auto shape = node->get_shape();
```

Comment:
`get_shape()` gets the shape from output 0 and gives an error if there's more than one. Also, `get_shape()` will get an error if it's a partial shape. So use `input_value(0).get_partial_shape()` and check that it's static and then get the shape from the partial shape.

Reply:
shape is out_shape, which we check for dynamic shape before the if. will use that here; will remove get_shape.